test_scripts/Defects/7_0/3164_1_OnInteriorVehicleData_empty_module_data.lua

```diff
@@ -0,0 +1,86 @@
+
+-- Steps:
+-- 1. HMI sends OnInteriorVehicleData notification with empty control data for <moduleType>
+-- SDL does:
+-- - a. merge received data with existed data in cache
+-- - b. send OnInteriorVehicleData notification with data received from HMI to mobile app
+-- 2. Mobile app requests GetInteriorVehicleData(<moduleType>)
+-- SDL does:
+-- a. respond to GetInteriorVehicleData with initial data from cache
+---------------------------------------------------------------------------------------------------
+-- [[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('user_modules/sequences/actions')
+local rc = require('user_modules/sequences/remote_control')
+
+-- [[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+config.application1.registerAppInterfaceParams.appHMIType = { "REMOTE_CONTROL" }
+
+-- [[ Local Variables ]]
+local appId = 1
+local isNotSubscribed = false
+local moduleData = {
+  emptyRadioData = { moduleType = "RADIO", radioControlData = {}},
+  emptyAudioData = { moduleType = "AUDIO", audioControlData = {}},
+  emptyHmiSettingsData = { moduleType = "HMI_SETTINGS", hmiSettingsControlData = {}},
+  emptyClimateData = { moduleType = "CLIMATE", climateControlData = {}},
+  emptySeatData = { moduleType = "SEAT", seatControlData = {}},
+}
+
+-- [[ Local Functions ]]
+local function getModuleId(pModuleType)
+  return rc.predefined.getModuleControlData(pModuleType, 1).moduleId
+end
+
+local function OnInteriorVehicleData(pModuleData)
+  local rpc = "OnInteriorVehicleData"
+  local mobSession = common.mobile.getSession(appId)
+  local hmi = common.hmi.getConnection()
+  hmi:SendNotification(rc.rpc.getHMIEventName(rpc), { moduleData = pModuleData })
+  mobSession:ExpectNotification(rc.rpc.getAppEventName(rpc), { moduleData = pModuleData })
+end
+
+local function GetInteriorVehicleData(pModuleData)
+  local rpc = "GetInteriorVehicleData"
+  local mobSession = common.mobile.getSession(appId)
+  local moduleId = getModuleId(pModuleData.moduleType)
+  pModuleData.moduleId = moduleId
+  local cid = mobSession:SendRPC(rc.rpc.getAppEventName(rpc),
+    rc.rpc.getAppRequestParams(rpc, pModuleData.moduleType, moduleId))
+  mobSession:ExpectResponse(cid, {
+    success = true,
+    resultCode = "SUCCESS",
+    moduleData = rc.state.getActualModuleIVData(pModuleData.moduleType, moduleId)
+  })
+end
+
+--[[Scenario]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", rc.rc.start)
+runner.Step("Register App", common.registerAppWOPTU)
+runner.Step("Activate App", common.activateApp)
+runner.Step("PTU with permission for RC", rc.rc.policyTableUpdate)
+for _, value in pairs (moduleData) do
+  runner.Step("Subscribe to moduleType " .. value.moduleType, rc.rc.subscribeToModule,
+    { value.moduleType, getModuleId(value.moduleType), appId, isNotSubscribed })
```

Comment:
@GetmanetsIrina `isNotSubscribed` is not reflecting purpose of its use here. Fourth parameter of `rc.rc.subscribeToModule` is `pIsSubscriptionCached`.
```suggestion
    { value.moduleType, getModuleId(value.moduleType), appId, isSubscriptionNotCached })
```